DiscordBot/Backends/Discord/DiscordBackend.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Threading.Tasks;
+using DSharpPlus.EventArgs;
+
+namespace DiscordBot.Backends.Discord
+{
+    class DiscordBackend: IBackend<MessageCreateEventArgs>
+    {
+        readonly Bot<MessageCreateEventArgs> _bot;
+
+        public DiscordBackend(Bot<MessageCreateEventArgs> bot) =>
+            _bot = bot;
+
+        public async Task Run()

```

Comment:
You can make this function get the bot instance, and it will make your life easier

Reply:
reminder to self: take the advice of the second sentence

Reply:
kinda resolved? i got the function but i still need to save it in a class variable, as it's used in a different method